steps/Proposals.tsx

```diff
@@ -0,0 +1,609 @@
+        console.log(`Selected option: ${option}`);
+    };
+
+    const handleSelect = (proposalId: number) => {
+        if (!toggleVoteActions) {
+            const selectedProposal = proposals.find((proposal) => proposal.proposalId === proposalId);
+            if (selectedProposal) {
+                if (selected && selected.proposalId === proposalId) {
+                    setSelected(null);
+                } else {
+                    setSelected(selectedProposal);
+                    const voteTrx = generateVoteTrx({
+                        proposalId: selectedProposal.proposalId,
+                        voterAddress: wallet.user!.address,
+                        option: selectedOption,
+                    });
+                    console.log(voteTrx);
+                }
+            }
+            console.log(proposalId.toString());
+        }
+    }
+
+    const toggelVotes = () => {
+        setToggleVoteActions(!toggleVoteActions)
+    }
+    const toggelVotesClose = () => {
+        setToggleVoteActions(false)
+    }
+
+    useEffect(() => {
+        const fetchProposals = async () => {
+            const proposalsRequest: QueryProposalsRequest = {
+                proposalStatus: "",
+                voter: "",
+                depositor: "",
+            }
+            try {
+                const response = await queryClient?.cosmos.gov.v1beta1.proposals(proposalsRequest);
+                if (response && response.proposals) {
+                    console.log(response.proposals);
+                    const proposalsArray: ProposalData[] = [];
+                    response.proposals.forEach((proposal) => {
+                        const proposalContentBytes = proposal.content?.value;
+                        const proposalType = proposal.content?.typeUrl.split('/')[1];
+                        if (proposalType === 'cosmos.gov.v1beta1.TextProposal') {
+                            const proposalContent = cosmos.gov.v1beta1.TextProposal.decode(proposalContentBytes);
+                            const proposalTitle = proposalContent.title;
+                            const proposalDescription = proposalContent.description;
+                            const proposer = proposal.proposer?.address || '';
+                            const proposalData: ProposalData = {
+                                proposer: proposer,
+                                title: proposalTitle,
+                                description: proposalDescription,
+                            };
+                            proposalsArray.push(proposalData);
+                        }
+                    });
+                    console.log(proposalsArray);
+                    setProposals(response.proposals);
+                }
+            } catch (error) {
+                console.log("Error fetching proposals:", error);
+            }
+        }
+        fetchProposals()
+    }, [])
+
+    const generateVoteTrx = ({
+        proposalId,
```

Comment:
@RafeeqWeideman-YDS this util should be with the other trx util functions in `/utils/transactions.tsx`